lib/node_modules/@stdlib/lapack/base/iladlr/README.md

```diff
@@ -0,0 +1,232 @@
+<!--
+
+@license Apache-2.0
+
+Copyright (c) 2024 The Stdlib Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# iladlr
+
+> Return last non-zero row of matrix `A`.
+
+<section class="usage">
+
+## Usage
+
+```javascript
+var iladlr = require( '@stdlib/lapack/base/iladlr' );
+```
+
+#### iladlr( order, M, N, A, LDA )
+
+Returns last non-zero row of matrix `A`.
+
+```javascript
+var Float64Array = require( '@stdlib/array/float64' );
+
+var out;
+var A = new Float64Array( [ 1.0, 2.0, 3.0, 4.0 ] );
+
+out = iladlr( 'row-major', 2, 2, A, 2 );
+// returns 1
+```
+
+The function has the following parameters:
+
+-   **order**: storage layout.
+-   **M**: number of rows in `A`.
+-   **N**: number of columns in `A`.
+-   **A**: input [`Float64Array`][mdn-float64array].
+-   **LDA**: stride of the first dimension of `A` (a.k.a., leading dimension of the matrix `A`).
+
+Note that indexing is relative to the first index. To introduce an offset, use [`typed array`][mdn-typed-array] views.
+
+<!-- eslint-disable stdlib/capitalized-comments -->
+
+```javascript
+var Float64Array = require( '@stdlib/array/float64' );
+
+// Initial an array.
+var A0 = new Float64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0 ] );
+
+// Create offset view.
+var A1 = new Float64Array( A0.buffer, A0.BYTES_PER_ELEMENT*1 ); // start at 2nd element
+
+var out = iladlr( 'row-major', 2, 2, A1, 2 );
+// out => 1
+```
+
+#### iladlr.ndarray( M, N, A, sa1, sa2, oa )
+
+Returns last non-zero row of matrix `A` using alternative indexing semantics.
+
+```javascript
+var Float64Array = require( '@stdlib/array/float64' );
+
+var out;
+var A = new Float64Array( [ 1.0, 2.0, 3.0, 4.0 ] );
+
+out = iladlr.ndarray( 2, 2, A, 2, 1, 0 );
+// returns 1
+```
+
+The function has the following parameters:
+
+-   **M**: number of rows in `A`.
+-   **N**: number of columns in `A`.
+-   **A**: input [`Float64Array`][mdn-float64array].
+-   **sa1**: stride of the first dimension of `A`.
+-   **sa2**: stride of the second dimension of `A`.
+-   **oa**: starting index for `A`.
+
+While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying buffer, the offset parameters support indexing semantics based on starting indices. For example,
+
+```javascript
+var Float64Array = require( '@stdlib/array/float64' );
+
+var out;
+var A = new Float64Array( [ 4.0, 3.0, 2.0, 4.0 ] );
+
+out = iladlr.ndarray( 2, 2, A, -2, -1, 3 );
+// returns 1
```

Comment:
Your examples all do the same thing. Here it is not obvious how the change in stride and offset have any bearing on the output.

In your examples, you should strive to make cause and effect more obvious.